core/src/main/java/org/ldaptive/control/util/SyncReplRunner.java

```diff
@@ -94,44 +65,136 @@ public void accept(final Exception e)
+   */
+  public static SingleConnectionFactory createConnectionFactory(final ConnectionConfig config)
   {
-    this(createTransport(), config, request, manager);
+    // CheckStyle:MagicNumber OFF
+    return createConnectionFactory(
+      config,
+      SearchConnectionValidator.builder()
+        .period(Duration.ofMinutes(1))
+        .timeout(Duration.ofSeconds(5))
+        .timeoutIsFailure(false)
+        .build());
+    // CheckStyle:MagicNumber ON
   }
 
 
   /**
-   * Creates a new sync repl runner.
+   * Creates a new single connection factory. See {@link #createTransport()}.
+   *
+   * @param  config  sync repl connection configuration
+   * @param  validator  connection validator
+   *
+   * @return  single connection factory for use with a sync repl runner
+   */
+  public static SingleConnectionFactory createConnectionFactory(
+    final ConnectionConfig config, final ConnectionValidator validator)
+  {
+    return createConnectionFactory(createTransport(), config, validator);
+  }
+
+
+  /**
+   * Creates a new single connection factory.
    *
    * @param  transport  sync repl connection transport
    * @param  config  sync repl connection configuration
-   * @param  request  sync repl search request
-   * @param  manager  sync repl cookie manager
+   * @param  validator  connection validator
+   *
+   * @return  single connection factory for use with a sync repl runner
    */
-  public SyncReplRunner(
-    final Transport transport,
-    final ConnectionConfig config,
-    final SearchRequest request,
-    final CookieManager manager)
+  public static SingleConnectionFactory createConnectionFactory(
+    final Transport transport, final ConnectionConfig config, final ConnectionValidator validator)
   {
-    connectionTransport = transport;
-    connectionConfig = config;
-    searchRequest = request;
-    cookieManager = manager;
+    final SingleConnectionFactory factory = new SingleConnectionFactory(config, transport);
+    factory.setValidator(validator);
+    configureConnectionFactory(factory);
+    return factory;
+  }
+
+
+  /**
+   * Configures the supplied factory for use with a {@link SyncReplRunner}. The factory's configuration will have the
+   * following modifications:
+   * <ul>
+   *   <li>{@link ConnectionConfig#setTransportOption(String, Object)} of AUTO_READ to false</li>
+   *   <li>{@link ConnectionConfig#setAutoReconnect(boolean)} to false</li>
+   *   <li>{@link ConnectionConfig#setAutoReplay(boolean)} to false</li>
+   *   <li>{@link SingleConnectionFactory#setFailFastInitialize(boolean)} to false</li>
+   *   <li>{@link SingleConnectionFactory#setNonBlockingInitialize(boolean)} to false</li>
+   *   <li>{@link AbstractConnectionValidator#setOnFailure(Consumer)} to
+   *   {@link SingleConnectionFactory.ReinitializeConnectionConsumer}</li>
+   * </ul>
+   *
+   * @param  factory  to configure
+   */
+  public static void configureConnectionFactory(final SingleConnectionFactory factory)
```

Comment:
This is the only static method that seems naturally at home in this class since the configuration behavior is tailed specifically to `SyncReplRunner`.